core/src/main/java/io/stargate/core/util/ByteBufferUtils.java

```diff
@@ -24,20 +24,28 @@
   private ByteBufferUtils() {}
 
   public static String toBase64(ByteBuffer buffer) {
-    return Base64.getEncoder().encodeToString(getArray(buffer));
+    return toBase64(getArray(buffer));
+  }
+
+  public static String toBase64(byte[] bytes) {
+    return Base64.getEncoder().encodeToString(bytes);
   }
 
   public static ByteBuffer fromBase64(String base64) {
     return ByteBuffer.wrap(Base64.getDecoder().decode(base64));
   }
 
   public static String toBase64ForUrl(ByteBuffer buffer) {
-    return Base64.getUrlEncoder().encodeToString(getArray(buffer));
+    return toBase64ForUrl(getArray(buffer));
+  }
+
+  public static String toBase64ForUrl(byte[] bytes) {
+    return Base64.getUrlEncoder().encodeToString(bytes);
   }
 
   public static ByteBuffer fromBase64UrlParam(String base64) {
     // TODO: remove support for legacy use cases when they are no longer relevant
```

Comment:
We can take care of this TODO now. The legacy case is no longer needed. It just needed to handle both while we were rolling out the change

https://github.com/stargate/stargate/pull/1041#discussion_r654583476

Reply:
Ah. In that case I will simplify this and remove specialized handling!